cddl/contrib/opensolaris/common/ctf/ctf_lookup.c

```diff
@@ -134,6 +134,8 @@ ctf_lookup_by_name(ctf_file_t *fp, const char *name)
 		for (lp = fp->ctf_lookups; lp->ctl_prefix != NULL; lp++) {
 			if (lp->ctl_prefix[0] == '\0' ||
 			    strncmp(p, lp->ctl_prefix, (size_t)(q - p)) == 0) {
```

Comment:
I wonder if this line should really have a `(size_t)(q - p) >= lp->ctl_len &&` for the RHS of the `||`. Bailing out inside the `if` seems inelegant, we shouldn't even be trying to do a partial match here, it makes no sense. The only reason to do a `strncmp` is because `p` has things beyond the prefix before its null terminator. Alternatively you could write it as `strncmp(p, lp->ctl_prefix, MAX(lp->ctl_len, (size_t)(q - p))`, which would make the `strncmp` itself return non-zero when `p` is too short. Either way, such a fix should probably go upstream to Illumos, though it appears FreeBSD forked long ago and isn't collaborating here any more?

Reply:
Thanks for the suggestions! I think that `(size_t)(q - p) >= lp->ctl_len &&` is the best way to go. 
I also opened a pr for upstream Illumos.

Reply:
well, `strncmp(p, lp->ctl_prefix, strlen(lp->ctl_prefix))` it's even better!